ui/dashboards/src/context/TemplateVariableProvider.tsx

```diff
@@ -0,0 +1,268 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { createContext, useContext } from 'react';
+import { createStore, useStore } from 'zustand';
+import { immer } from 'zustand/middleware/immer';
+import { devtools } from 'zustand/middleware';
+import { TemplateVariableContext } from '@perses-dev/plugin-system';
+import {
+  VariableDefinitions,
+  VariablesState,
+  VariableState,
+  VariableName,
+  VariableValue,
+  ListVariableDefinition,
+  VariableDefinition,
+  DEFAULT_ALL_VALUE as ALL_VALUE,
+} from '@perses-dev/core';
+
+type TemplateVariableSrv = {
```

Comment:
what does `srv` stand for?

Reply:
Lol yeah that's not super clear, probably need a better name. It means `service` which is used in angular (aka Grafana). Is `TemplateVariableState` or like `TemplateVariableStateManager` better?

Reply:
Both sound good to me!